### PR TITLE
Add ability to define custom entrypoints in async

### LINKED ISSUE #1797: Add customizable handle signal in async logic
### Problem to Solve

After #1730 

User may need to clean resources when signal occurred in async implicit logic

```rust
#[gstd::async_main(handle_signal = handle_signal)]
async fn main() {
    // ...
}

fn handle_signal() {
    gstd::handle_signal();
    // clean some resources
}

```

### Possible Solution

.

### Notes

_No response_

### PR DESCRIPTION
**Pull request by ark0f**
Resolves #1797

### CODE REVIEW

**Review by shamilsan — APPROVED** [review]
Looks awesome 🔥

File: gstd/codegen/tests/ui/double_definition_not_work.rs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#![no_main]
+
+#[gstd::async_init(handle_signal = custom_handle_signal)]
```
Comment (breathx) [inline gstd/codegen/tests/ui/double_definition_not_work.rs:3]: similar tests for handle_reply?

File: examples/binaries/async-custom-entry/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            USER = msg::source();
+        }
+    }
+
+    #[gstd::async_main]
+    async fn main() {
+        loop {}
+    }
+
+    fn my_handle_reply() {
```
Comment (breathx) [inline examples/binaries/async-custom-entry/src/lib.rs:47]:
```suggestion
    fn _handle_reply() {
```
What about such thing?

File: gstd/codegen/src/lib.rs [hunk 3]
```diff
@@ -123,13 +206,33 @@ fn generate_if_required(code: TokenStream) -> TokenStream {
 ///     gstd::debug!("Hello world!");
 /// }
 /// ```
+///
+/// You can specify `handle_reply` and `handle_signal` using parameters of same names:
```
Comment (breathx) [inline gstd/codegen/src/lib.rs:210]: Expand docs that default behaviour kept

File: gstd/codegen/tests/ui/double_definition_not_work.stderr [hunk 4]
```diff
@@ -0,0 +1,5 @@
+error: parameter already defined
+ --> tests/ui/double_definition_not_work.rs:6:36
```
Comment (breathx) [inline gstd/codegen/tests/ui/double_definition_not_work.stderr:2]: should we keep stderr file?

**Review by breathx — APPROVED** [review]

File: gstd/codegen/tests/ui/double_definition_not_work.stderr (hunk 4 above)
Comment (ark0f) [reply gstd/codegen/tests/ui/double_definition_not_work.stderr]: Yes. It's for 	`trybuild`

**Pull request merged**